File: android/src/main/kotlin/com/usercentrics/sdk/flutter/bridge/GetCMPDataBridge.kt
package com.usercentrics.sdk.flutter.bridge

import com.usercentrics.sdk.flutter.api.FlutterMethodCall
import com.usercentrics.sdk.flutter.api.FlutterResult
import com.usercentrics.sdk.flutter.api.UsercentricsProxy
import com.usercentrics.sdk.flutter.api.UsercentricsProxySingleton
import com.usercentrics.sdk.flutter.serializer.serialize

internal class GetCMPDataBridge(
    private val usercentrics: UsercentricsProxy = UsercentricsProxySingleton
) : MethodBridge {
    override val name: String
        get() = "getCMPData"

    override fun invoke(call: FlutterMethodCall, result: FlutterResult) {
        assert(name == call.method)
        result.success(usercentrics.instance.getCMPData().serialize())
    }
}

File: ios/usercentrics_sdk/Sources/usercentrics_sdk/Bridge/GetCMPDataBridge.swift
import Usercentrics
import Flutter

struct GetCMPDataBridge : MethodBridge {
    let name: String = "getCMPData"
    let usercentrics: UsercentricsProxyProtocol

    func invoke(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
        assert(call.method == name)
        let cmpData = usercentrics.shared.getCMPData()
        result(cmpData.serialize())
    }
}
